PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:15 + add:0, change matches:
-    _inherit = ['account.move']
-    _inherit = ["account.move.line"]
-    _inherit = ['product.template']
-    _inherit = ['res.company']
-    _inherit = ['res.config.settings']
-    _inherit = ['res.users']
-    _inherit = ["sale.order"]
-    _inherit = ['sale.order.line']
-    _inherit = ["stock.route"]
-    _inherit = ["stock.move"]
-    _inherit = ["stock.move.line"]
-    _inherit = ['procurement.group']
-    _inherit = ['stock.rule']
-    _inherit = ['stock.picking']
-    _inherit = ['stock.lot']

Total Changes: 30

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/sale_stock/models/account_move.py
+++ b/addons/sale_stock/models/account_move.py
@@ -9,7 +9,7 @@ from odoo.tools.misc import formatLang
 
 
 class AccountMove(models.Model):
-    _inherit = ['account.move']
+    _inherit = 'account.move'
 
     def _stock_account_get_last_step_stock_moves(self):
         """ Overridden from stock_account.
@@ -142,7 +142,7 @@ class AccountMove(models.Model):
 
 
 class AccountMoveLine(models.Model):
-    _inherit = ["account.move.line"]
+    _inherit = "account.move.line"
 
     def _sale_can_be_reinvoice(self):
         self.ensure_one()

--- a/addons/sale_stock/models/product_template.py
+++ b/addons/sale_stock/models/product_template.py
@@ -5,7 +5,7 @@ from odoo import api, models
 
 
 class ProductTemplate(models.Model):
-    _inherit = ['product.template']
+    _inherit = 'product.template'
 
     @api.depends('type')
     def _compute_expense_policy(self):

--- a/addons/sale_stock/models/res_company.py
+++ b/addons/sale_stock/models/res_company.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = ['res.company']
+    _inherit = 'res.company'
 
     security_lead = fields.Float(
         'Sales Safety Days', default=0.0, required=True,

--- a/addons/sale_stock/models/res_config_settings.py
+++ b/addons/sale_stock/models/res_config_settings.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     security_lead = fields.Float(related='company_id.security_lead', string="Security Lead Time", readonly=False)
     use_security_lead = fields.Boolean(

--- a/addons/sale_stock/models/res_users.py
+++ b/addons/sale_stock/models/res_users.py
@@ -5,7 +5,7 @@ from odoo import models, fields
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     property_warehouse_id = fields.Many2one('stock.warehouse', string='Default Warehouse', company_dependent=True, check_company=True)
 

--- a/addons/sale_stock/models/sale_order.py
+++ b/addons/sale_stock/models/sale_order.py
@@ -12,7 +12,7 @@ _logger = logging.getLogger(__name__)
 
 
 class SaleOrder(models.Model):
-    _inherit = ["sale.order"]
+    _inherit = "sale.order"
 
     incoterm = fields.Many2one(
         'account.incoterms', 'Incoterm',

--- a/addons/sale_stock/models/sale_order_line.py
+++ b/addons/sale_stock/models/sale_order_line.py
@@ -11,7 +11,7 @@ from odoo.exceptions import UserError
 
 
 class SaleOrderLine(models.Model):
-    _inherit = ['sale.order.line']
+    _inherit = 'sale.order.line'
 
     qty_delivered_method = fields.Selection(selection_add=[('stock_move', 'Stock Moves')])
     route_id = fields.Many2one('stock.route', string='Route', domain=[('sale_selectable', '=', True)], ondelete='restrict')

--- a/addons/sale_stock/models/stock.py
+++ b/addons/sale_stock/models/stock.py
@@ -8,12 +8,12 @@ from odoo.tools.sql import column_exists, create_column
 
 
 class StockRoute(models.Model):
-    _inherit = ["stock.route"]
+    _inherit = "stock.route"
     sale_selectable = fields.Boolean("Selectable on Sales Order Line")
 
 
 class StockMove(models.Model):
-    _inherit = ["stock.move"]
+    _inherit = "stock.move"
     sale_line_id = fields.Many2one('sale.order.line', 'Sale Line', index='btree_not_null')
 
     @api.model
@@ -58,20 +58,20 @@ class StockMove(models.Model):
 
 
 class StockMoveLine(models.Model):
-    _inherit = ["stock.move.line"]
+    _inherit = "stock.move.line"
 
     def _should_show_lot_in_invoice(self):
         return 'customer' in {self.location_id.usage, self.location_dest_id.usage}
 
 
 class ProcurementGroup(models.Model):
-    _inherit = ['procurement.group']
+    _inherit = 'procurement.group'
 
     sale_id = fields.Many2one('sale.order', 'Sale Order')
 
 
 class StockRule(models.Model):
-    _inherit = ['stock.rule']
+    _inherit = 'stock.rule'
 
     def _get_custom_move_fields(self):
         fields = super(StockRule, self)._get_custom_move_fields()
@@ -80,7 +80,7 @@ class StockRule(models.Model):
 
 
 class StockPicking(models.Model):
-    _inherit = ['stock.picking']
+    _inherit = 'stock.picking'
 
     sale_id = fields.Many2one('sale.order', compute="_compute_sale_id", inverse="_set_sale_id", string="Sales Order", store=True, index='btree_not_null')
 
@@ -201,7 +201,7 @@ class StockPicking(models.Model):
 
 
 class StockLot(models.Model):
-    _inherit = ['stock.lot']
+    _inherit = 'stock.lot'
 
     sale_order_ids = fields.Many2many('sale.order', string="Sales Orders", compute='_compute_sale_order_ids')
     sale_order_count = fields.Integer('Sale order count', compute='_compute_sale_order_ids')
